opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/AppResources.java

```diff
@@ -105,112 +134,174 @@ public Response getApp(@QueryParam("appid") Long appId)
 	@Produces(MediaType.APPLICATION_JSON)
 	@RolesAllowed({AuthorizationCheck.ODCS_API_ADMIN, AuthorizationCheck.ODCS_API_USER})
 	public Response postApp(ApiLoadingApp app)
-		throws WebAppException, DbException, SQLException
+			throws DbException
 	{
-		LOGGER.debug("Post app received app {} with id {}", app.getAppName(), app.getAppId());
-		try (DbInterface dbi = new DbInterface();
-			ApiAppDAO dao = new ApiAppDAO(dbi))
+		try (LoadingAppDAI dai = getLegacyDatabase().makeLoadingAppDAO())
 		{
-			dao.writeApp(app);
-			return ApiHttpUtil.createResponse(app);
+			dai.writeComputationApp(map(app));
+			return Response.status(HttpServletResponse.SC_OK)
+					.entity(String.format("Wrote app to database with ID: %s", app.getAppId()))
+					.build();
+		}
+		catch(DbIoException ex)
+		{
+			throw new DbException("Unable to store app", ex);
 		}
 	}
 
+	static CompAppInfo map(ApiLoadingApp app)
+	{
+		CompAppInfo ret = new CompAppInfo();
+		if (app.getAppId() != null)
+		{
+			ret.setAppId(DbKey.createDbKey(app.getAppId()));
```

Comment:
I think `DbKey.NullKey` should be used if the app id is null

Reply:
Added `NullKey` as result if app id is null